### PR TITLE
drivers/mtd: Add check for interger overflow

### PR DESCRIPTION
**Pull request by Teufelchen1**
### Contribution description

Hei 🦫

Fixes a minor potential for integer overflow.

### Testing procedure

CI / good review
Maybe `make -C tests/unittests tests-mtd test`

### CODE REVIEW

File: drivers/mtd_emulated/mtd_emulated.c [hunk 1]
```diff
@@ -16,6 +16,7 @@
 static int _init(mtd_dev_t *dev)
 {
     mtd_emulated_t *mtd = (mtd_emulated_t *)dev;
+
```
Comment (Teufelchen1) [inline drivers/mtd_emulated/mtd_emulated.c:19]: Unrelated change, sneaking it in to make the file happy in uncrustify.

**Comment by Teufelchen1** [comment]
Switched from the unsigned buildins to the "you will figure the type out" buildins.

**Review by maribu — APPROVED** [review]

**Comment by maribu** [comment]
> /opt/riot-toolchain/msp430-elf/10.1.0-18/bin/../lib/gcc/msp430-elf/10.1.0/../../../../msp430-elf/bin/ld: section .bss LMA wraps around address space
> collect2: error: ld returned 1 exit status

This means it got larger than what would fit into 16 bit address space, so it needs to be added to `Makefile.ci`.

Btw.: I get this a lot with the `BUILD_IN_DOCKER=1` toolchain, and rarely with Alpine's toolchain. My guess is that the newer linker will perform the check if data exceeds the 16 bit address space *after* the garbage collection phase, and the older one did this *before*. This would explain why the same app build on my toolchain not only ends up fitting into 16 bit address space, but also into the (smaller) RAM and ROM (and behave as expected), while it fails in the CI.

But anyway, bumping `Makefile.ci` it is for now and if the toolchain gets updated, we can re-enable linking in the CI of that app.
